django/db/models/expressions.py

```diff
@@ -538,6 +547,200 @@ def get_group_by_cols(self):
         return [self]
 
 
+class BaseCaseExpression(ExpressionNode):
+    """
+    Base class for all case expressions.
+    """
+    NoDefault = object()
+
+    def __init__(self, cases=None, default=NoDefault, output_field=None):
+        super(BaseCaseExpression, self).__init__(output_field)
+        self.cases = self._parse_cases(cases)
+        if default is self.NoDefault:
+            default = None
```

Comment:
If you assign the default as `Value(None)`, then you shouldn't need to conditionally add `self.default` in get/set_source_expressions. It should just convert to NULL sql string.